### DISCUSSION TITLE
Theme Support

### DISCUSSION THREAD

**Discussion by christianvozar** [iniquitybbs/iniquity#23] [Ideas]
## Overview

Themes/Menu Sets/Skins for BBS development has always been cumbersome for artists, typically not programmers, to create and distribute for general sysop use. Sysops should feel if a theme is distributed they can install it with appropriate license/permission to do so; Artists should feel a theme will be displayed unaltered (no art modifications, stripping of artist name).

## Proposal

Container format for themes for IQ3. Simple compressed file (`.zip`), Manifest of contained artwork, artist name, group name, version, theme name, theme description (possibly as .json or .yaml). Optional file_id.diz can be included for thumbnail reference when released in a repo.

IQ3 should perform validation on theme pack to ensure `required` art is included prior to install/use by sysop.

**Comment by ispyhumanfly** [comment]
@christianvozar I think something along these lines is a pretty fantastic idea. I agree that the problem is real, and, this new effort to re-invent the BBS is an opportunity to address it. I like some of the concepts you just outlined, too...

Right now we're at that point where we're super close to needing to define such cases and develop this feature more fully.

**Comment by lmanuel** [comment]
I'll start working on some custom artwork for the iNiQUiTY font logo and opening login screen.

**Comment by ispyhumanfly** [comment]
I'm hopeful that where we are headed with @iniquitybbs/templates and @iniquitybbs/themes packages will satisfy all or part of the ideas captured here. One thing we could start doing is using the iniquity.json file as this manifest. All artwork used for a template/package can be validated anytime someone initializes a new iniquity bbs with one of those templates/themes.

**Comment by ispyhumanfly** [comment]
This discussion will remain as issue #38
